source/simulator/introspection.cc

```diff
@@ -487,6 +524,22 @@ namespace aspect
   }
 
 
+
+  template <int dim>
+  bool
+  Introspection<dim>::is_composition_component (const unsigned int component_index) const
+  {
+    // All compositions live at the end. Just to be sure, there are no other components
+    // in our system after compositional fields, right?
```

Comment:
Does that work in models with melt transport? We do have the melt velocity, fluid pressure etc. before the temperature, right?

Reply:
Yes, that is correct. But I added an Assert here just in case.